src/web_buffer.h

```diff
@@ -1,77 +1,266 @@
 #ifndef NETDATA_WEB_BUFFER_H
 #define NETDATA_WEB_BUFFER_H 1
 
-#define WEB_DATA_LENGTH_INCREASE_STEP 1024
+/**
+ * @file web_buffer.h
+ * @brief Data structure and API of web_buffer.
+ *
+ * A web buffer is a buffer to store data for sending or recieving on web clients or servers.
+ * A web buffer increases it's size automatic.
+ * A web buffer is always null terminated.
+ */
 
+#define WEB_DATA_LENGTH_INCREASE_STEP 1024 ///< Minimum size a web buffer gets increased.
+
+/** web buffer */
 typedef struct web_buffer {
-    size_t size;        	// allocation size of buffer, in bytes
-    size_t len;     		// current data length in buffer, in bytes
-    char *buffer;   		// the buffer itself
-    uint8_t contenttype;	// the content type of the data in the buffer
-    uint8_t options;		// options related to the content
-    time_t date;    		// the timestamp this content has been generated
-    time_t expires;			// the timestamp this content expires
+    size_t size;        	///< Total size of `buffer`, in bytes.
+    size_t len;     		///< Used size of `buffer`, in bytes.
+    char *buffer;   		///< The conetent
+    uint8_t contenttype;	///< Content type of the buffer. (CT_*)
+    uint8_t options;		///< WB_CONTENT_*
+    time_t date;    		///< Timestamp the content has been created.
+    time_t expires;			///< Timestamp the content expires.
 } BUFFER;
 
 // options
-#define WB_CONTENT_CACHEABLE            1
-#define WB_CONTENT_NO_CACHEABLE         2
+#define WB_CONTENT_CACHEABLE            1 ///< ktsaou: Your help needed
+#define WB_CONTENT_NO_CACHEABLE         2 ///< ktsaou: Your help needed
 
 // content-types
-#define CT_APPLICATION_JSON             1
-#define CT_TEXT_PLAIN                   2
-#define CT_TEXT_HTML                    3
-#define CT_APPLICATION_X_JAVASCRIPT     4
-#define CT_TEXT_CSS                     5
-#define CT_TEXT_XML                     6
-#define CT_APPLICATION_XML              7
-#define CT_TEXT_XSL                     8
-#define CT_APPLICATION_OCTET_STREAM     9
-#define CT_APPLICATION_X_FONT_TRUETYPE  10
-#define CT_APPLICATION_X_FONT_OPENTYPE  11
-#define CT_APPLICATION_FONT_WOFF        12
-#define CT_APPLICATION_FONT_WOFF2       13
-#define CT_APPLICATION_VND_MS_FONTOBJ   14
-#define CT_IMAGE_SVG_XML                15
-#define CT_IMAGE_PNG                    16
-#define CT_IMAGE_JPG                    17
-#define CT_IMAGE_GIF                    18
-#define CT_IMAGE_XICON                  19
-#define CT_IMAGE_ICNS                   20
-#define CT_IMAGE_BMP                    21
-#define CT_PROMETHEUS                   22
+#define CT_APPLICATION_JSON             1  ///< MIME Type application/json
+#define CT_TEXT_PLAIN                   2  ///< MIME Type text/plain
+#define CT_TEXT_HTML                    3  ///< MIME Type text/html
+#define CT_APPLICATION_X_JAVASCRIPT     4  ///< MIME Type application/javascript. \todo We should switch this to CT_APPLICATION_JAVASCRIPT
```

Comment:
I did some research which are the correct types. I have no motivation to fix this now. If you think this may be fixed I leave the todos in. Check the other types too...